### ISSUE TITLE
Sync to iOS shows company instead of person name

### ISSUE THREAD

**Issue by tomasjay** [nextcloud/contacts#3603]
### Describe the bug

When creating a contact on NextCloud in the browser, it is synced to my iPhone, showing the company field as the contact's name. The contact's name is not visible in iOS at all, so basically the contact's name is replaced with the company field (see example "Bert Böller" below. 
Contacts with empty company field also sync correctly to iOS. 
And the other way around also works fine: Creating a contact on iOS including a company name will be synced to NextCloud without problems (example "Anton Aberwitz"). 

### Steps to reproduce

1. Setup contact sync with your iPhone according to the description here: https://docs.nextcloud.com/server/19/benutzerhandbuch/pim/sync_ios.html
2. Create a contact in NextCloud, using the browser. Include a company name as in example "Bert Böller".
<img width="1190" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/8744eb8c-35d9-473d-926b-03e986d17cce">

4. Drag down the screen on your iPhone to updated contacts.
5. See the result.
![image](https://github.com/nextcloud/contacts/assets/3900727/c9aa4932-0c06-483d-911b-52bc26d7c6de)
![image](https://github.com/nextcloud/contacts/assets/3900727/6ddaa774-8f11-49c7-aceb-d068329bd3bd)

6. Create a contact on iOS, including a company name, just like this:
![image](https://github.com/nextcloud/contacts/assets/3900727/dd30212b-9529-49e3-a5bf-ae662b7cc358)

7. Sync and see the result on NextCloud in the browser: It's fine. Please note that there is a "Detailed name" section which is not visible with the contact created on NextCloud. 
<img width="1189" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/e95731ca-bd41-43fd-9158-e6d3af125cf3">

### Expected behavior

The contact's person name should show up in contacts on iOS.
As a workaround: The contact's "Details name" section should be visible in NextCloud, so the user can decide how the person name will be synced.  

### Actual behavior

The contact's company field shows up as the person name in iOS. The person name is not synced at all.
The contact's "Detail name" section is not visible on NextCloud.

### Contact version

5.3.2

### Operating system

Linux 5.4.0-156-generic x86_64

### PHP engine version

Other

### Web server

Nginx

### Database

MySQL

### Additional info

PHP version is 8.2.8.
We are using a NextCloud installation provided by IONOS. According to a lookup at sitereport.netcraft.com they use Nginx as webserver.  
MySQL version is 10.5.19

**Comment by KM0376** [comment]
Same here. I can confirm

**Comment by ChristophWurst** [comment]
export the vcs and post the contents here please

**Comment by KM0376** [comment]
Export from NC web frontend:

BEGIN:VCARD
VERSION:3.0
PRODID:-//Sabre//Sabre VObject 4.4.2//EN
UID:4336f460-1f72-450e-80de-13cd422ce39a
FN:Tobias Test
ADR;TYPE=HOME:;;;;;;
EMAIL;TYPE=HOME:
TEL;TYPE=HOME,VOICE:
ORG:Companyname
REV;VALUE=DATE-AND-OR-TIME:20230911T082129Z
END:VCARD

Export form macOS adress book:

BEGIN:VCARD
VERSION:3.0
PRODID:-//Apple Inc.//macOS 13.5.2//EN
N:;;;;
FN:Companyname
ORG:Companyname;
X-ABShowAs:COMPANY
END:VCARD

**Comment by tomasjay** [comment]
### Workaround: 

You can add the fields `Detailed name` to the contact by selecting from the button `+ Add more info` > `... More fields` > `Detailed name`. After filling in, the contact syncs to iOS with the proper name. See example "Bert Böller".

1. Add data in `Detailed name`. In this example I filled in "Bert (Detail)" for `First name` and "Böller (Detail)" for `Last name`. 
![image](https://github.com/nextcloud/contacts/assets/3900727/91b37e99-9478-48be-9ac4-cc185774d5a1)

2. Sync: The contact name "BBB GmbH" is replaced by "Bert (Detail) Böller (Detail)".
![image](https://github.com/nextcloud/contacts/assets/3900727/f2949f17-aef6-40a7-8d6e-1d91a663660c)

However, the field `Name` of the contact does not update automatically in NextCloud. So, if you generate a new contact this way, you have to fill in both, the field `Name` and `Detailed name`. Else you will find your contact in NextCloud listed as "Name" as in the screenshot below (should be "Doris Dürr"). I think, there is a thread about this issue as well.

1. Add data to `Detailed name`.
<img width="1191" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/1201f04f-d63f-4d21-a312-55a3c646ed75">

2. After saving, the `Name` field shows "Name". 
<img width="1191" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/400984f3-1722-4c10-ae43-ad5a0ea8e66b">

This can also be worked around by changing the display settings from `Display name` to e.g. `Last name` (you'll have to refresh the browser window). 

<img width="866" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/ef4aa1a8-02bb-4b03-b1dc-a2f292d5e7cb">

**And!** – There is a hidden function to copy the `Detailed name` data to the `Name` field by clicking on `...` close to `Detailed name`. 

<img width="1197" alt="image" src="https://github.com/nextcloud/contacts/assets/3900727/e8fedb3f-d208-4e95-8d8a-f36d6e32735c">

### Suggested solution: 

So, basically, the field `Name` aka `Display name` aka `Full name` has no meaning for syncing via CardDAV. And for having a proper heading for the contact in the web, you'll have to take some extra steps which are not documented. 

Ideally, there would be a function that updates the `Name` field automatically from data in `Detailed name`. 

Also, there should be a clear and consistent naming for the `Name`field. In some places it is referred to as `Full name` or `Display name`.

Until then, please add a proper description to the NextCloud documentation, as probably many users might have the same issue with CardDAV sync.

**Comment by tomasjay** [comment]
Here is the other issue: Saving "Name" as default in the `Name` field: https://github.com/nextcloud/contacts/issues/3554

**Comment by tomasjay** [comment]
Any news on this? As mentioned above, this is not really a bug, but an unexpected behaviour that would need documentation in the user manual. Please confirm that this behaviour will be included in the user documentation for the contact app.
